codice/ddf
# DDF-5394 Allow User-Settings to be extensible via extension points

#### What does this PR do?
Allow User-Settings to be extensible via extension points. Also ensures that downstream projects that use plugins instead can still extend User-Settings via plugins.

#### Who is reviewing it? 
@djblue 
@andrewzimmer 
@willwill96 
@vinamartin 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@djblue
@vinamartin

#### How should this be tested?
Build DDF, make sure there is no regression in the User-Settings.
Additionally, this PR should NOT be merged in until we can confirm in downstream projects that no regressions have taken place in User-Settings

#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: #____

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
